ios/CodePush/CodePushPackage.m

```diff
@@ -292,26 +292,37 @@ + (NSString *)getCodePushPath
 
 + (NSDictionary *)getCurrentPackage:(NSError **)error
 {
-    NSString *packageHash = [CodePushPackage getCurrentPackageHash:error];
-    if (*error || !packageHash) {
+    NSError *localError;
+    NSString *packageHash = [CodePushPackage getCurrentPackageHash:&localError];
+    if (!packageHash) {
+        if (localError && error) {
+            *error = localError;
+        }
         return nil;
     }
-    
+
     return [CodePushPackage getPackage:packageHash error:error];
 }
 
 + (NSString *)getCurrentPackageBundlePath:(NSError **)error
 {
-    NSString *packageFolder = [self getCurrentPackageFolderPath:error];
+    NSError *localError;
```

Comment:
If possible, I would like to get rid of the need to create this new `localError` entirely, here and the rest of your changes, unless they serve some other purpose which I am missing.